daemon/cmd/kube_proxy_replacement.go

```diff
@@ -449,6 +415,45 @@ func finishKubeProxyReplacementInit(isKubeProxyReplacementStrict bool) {
 
 	// After this point, BPF NodePort should not be disabled
 
+	if !option.Config.EnableHostLegacyRouting {
```

Comment:
Maybe it makes sense to make a separate commit to move this entire block, then a new commit adding the change you want, so that we don't lose the history / break git bisect?